### ISSUE TITLE
TypeError: Cannot convert undefined or null to object  at Function.keys

### ISSUE THREAD

**Issue by benhinchley** [grpc/grpc-node#790]
### Problem description
Creating a new `Client` in grpc@1.19.0, I run into the following error

```
/code.sajari.com/sdk-node/node_modules/grpc/src/client.js:383
  Object.keys(self.$method_definitions).forEach((method_name) => {
         ^

TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at new Client (/code.sajari.com/sdk-node/node_modules/grpc/src/client.js:383:10)
    at new APIClient (/code.sajari.com/sdk-node/dist/api.js:51:23)
    at new Client (/code.sajari.com/sdk-node/dist/client.js:10:23)
    at Object.<anonymous> (/code.sajari.com/sdk-node/bin/v2.js:11:16)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
```

I have gone through the git history and it appears to have changed in this commit
https://github.com/grpc/grpc-node/commit/31879f85614fc27f3cb400345055d24c1679e26d

I have currently resolved this by reverting the grpc package back to version 1.16.1

### Environment
 - OS name, version and architecture: macOS Sierra v10.12.6
 - Node version: v10.9.0
 - Node installation method: homebrew
 - Package name and version: gRPC@1.19.0

**Comment by murgatroid99** [comment]
Can you share your code or something that reproduces this error? We haven't seen this in any of our tests or from anyone else, so there must be something specific about the way you are interacting with the library that is triggering this error.

**Comment by benhinchley** [comment]
The code that I was using to get the error is this https://github.com/sajari/sdk-node/blob/d770c95c633cfeba0d9efd371eb97ec82c8e56cf/src/api.ts#L84 

It did seem off to me that I couldn't find another issue on the matter.

**Comment by murgatroid99** [comment]
Oh, I see what the problem is. In previous versions we were iterating over that variable in a way that properly handled `null` and `undefined` values. I can have a fix for this in the next patch.

**Comment by brandonros** [comment]
@murgatroid99 It'd be a massive help if there was some kind of example code for how to make a call with a generic Client not using protobufs

**Comment by Bhagyashree39** [comment]
I'm also having same error but still it is not running. I have installed grpc package also but no solution. It is giving same error for all. What is the solution for this now?

**Comment by brandonros** [comment]
Update to 1.20.0

**Comment by lzambarda** [comment]
I am now getting a very similar error while running grpc@1.22.0 and parsing a received gRPC message:

```
TypeError: Cannot convert undefined or null to object
     at Function.keys (<anonymous>)
     at Function.Metadata._fromCoreRepresentation (.../node_modules/grpc/src/metadata.js:232:12)
     at handleNewCall (.../node_modules/grpc/src/server.js:760:29)
```

**Comment by benhinchley** [comment]
Closing this as this is now fixed as of #799 and #830 being merged
